src/_gettsim/interface.py

```diff
@@ -526,24 +533,33 @@ def _fail_if_group_variables_not_constant_within_groups(
     ----------
     data_tree
         Nested dictionary with pandas.Series as leaf nodes.
+    functions_tree
+        Nested dictionary with PolicyFunction as leaf nodes.
     """
-    # Extract group IDs from the 'groupings' branch.
-    grouping_ids_in_data_tree = data_tree.get("groupings", {})
+    group_by_functions_tree = flatten_dict.unflatten(
+        {
+            path: func
+            for path, func in flatten_dict.flatten(functions_tree).items()
+            if isinstance(func, GroupByFunction)
+        }
+    )
+    flat_data_tree = flatten_dict.flatten(data_tree)
 
     def faulty_leaf(path, leaf):
-        leaf_name = path[-1]
-        for grouping in SUPPORTED_GROUPINGS:
-            id_name = f"{grouping}_id"
-            if leaf_name.endswith(grouping) and id_name in grouping_ids_in_data_tree:
-                # Retrieve the corresponding group ID series from the data tree.
-                group_id_series = grouping_ids_in_data_tree.get(id_name)
+        group_by_id = get_group_by_id_path(
```

Comment:
Me changing the behavior of `shared.get_group_by_id_path` led to lots of errors here. Let's discuss that, I am not perfectly sure yet what is the best way out.